packages/@aws-cdk/aws-synthetics/test/canary.test.ts

```diff
@@ -0,0 +1,251 @@
+      's3:GetBucketLocation',
+      's3:ListAllMyBuckets',
+      'cloudwatch:PutMetricData',
+      'logs:CreateLogGroup',
+      'logs:CreateLogStream',
+      'logs:PutLogEvents',
+    ],
+  }));
+
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'mycanary',
+    role,
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Name: 'mycanary',
+    Code: {
+      Handler: 'index.handler',
+      Script: 'exports.handler = async () => {\nconsole.log(\'hello world\');\n};',
+    },
+    RuntimeVersion: 'syn-1.0',
+    ExecutionRoleArn: objectLike({
+      'Fn::GetAtt': arrayWith('roleC7B7E775'),
+    }),
+  });
+});
+
+test('Canary can have specified s3 Bucket', () => {
+  // GIVEN
+  const bucket = new s3.Bucket(stack, 'mytestbucket');
+
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'mycanary',
+    artifactBucket: bucket,
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Name: 'mycanary',
+    Code: {
+      Handler: 'index.handler',
+      Script: 'exports.handler = async () => {\nconsole.log(\'hello world\');\n};',
+    },
+    RuntimeVersion: 'syn-1.0',
+    ArtifactS3Location: objectLike({
+      'Fn::Join': arrayWith(arrayWith('s3://', {Ref: 'mytestbucket8DC16178'})),
+    }),
+  });
+});
+
+test('Canary can set schedule with Rate', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'mycanary',
+    schedule: synthetics.Schedule.rate(Duration.minutes(3)),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Name: 'mycanary',
+    Code: {
+      Handler: 'index.handler',
+      Script: 'exports.handler = async () => {\nconsole.log(\'hello world\');\n};',
+    },
+    RuntimeVersion: 'syn-1.0',
+    Schedule: objectLike({ Expression: 'rate(3 minutes)'}),
+  });
+});
+
+test('Canary can set schedule with Expression', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'mycanary',
+    schedule: synthetics.Schedule.expression('rate(3 minutes)'),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Name: 'mycanary',
+    Code: {
+      Handler: 'index.handler',
+      Script: 'exports.handler = async () => {\nconsole.log(\'hello world\');\n};',
+    },
+    RuntimeVersion: 'syn-1.0',
+    Schedule: objectLike({ Expression: 'rate(3 minutes)'}),
+  });
+});
+
+test('Canary can set schedule to run once', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'mycanary',
+    schedule: synthetics.Schedule.once(),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Name: 'mycanary',
+    Code: {
+      Handler: 'index.handler',
+      Script: 'exports.handler = async () => {\nconsole.log(\'hello world\');\n};',
+    },
+    RuntimeVersion: 'syn-1.0',
+    Schedule: objectLike({ Expression: 'rate(0 minutes)'}),
+  });
```

Comment:
```suggestion
  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
    Schedule: objectLike({ Expression: 'rate(0 minutes)'}),
  });
```

Also frankly - this is not your fault but I think this is a terrible way to specify "once"... #meh